homeassistant/helpers/template.py

```diff
@@ -44,22 +62,11 @@ def render(hass, template, variables=None, **kwargs):
     if variables is not None:
         kwargs.update(variables)
 
-    location_methods = LocationMethods(hass)
-    utcnow = dt_util.utcnow()
-
     try:
-        return ENV.from_string(template, {
-            'closest': location_methods.closest,
-            'distance': location_methods.distance,
-            'float': forgiving_float,
-            'is_state': hass.states.is_state,
-            'is_state_attr': hass.states.is_state_attr,
-            'now': dt_util.as_local(utcnow),
-            'states': AllStates(hass),
-            'utcnow': utcnow,
-            'as_timestamp': dt_util.as_timestamp,
-            'relative_time': dt_util.get_age
-        }).render(kwargs).strip()
+        if not isinstance(template, jinja2.Template):
+            template = compile_template(hass, template)
```

Comment:
Enforcing it will cause people to write better optimized code, so that's good. Let's remove it once every piece of code that renders templates has been converted 👍